packages/kit/test/apps/basics/test/cross-platform/client.test.js

```diff
@@ -221,15 +221,22 @@ test.describe('Navigation lifecycle functions', () => {
 	});
 
 	test('beforeNavigate is triggered after clicking a download link', async ({ page, baseURL }) => {
-		await page.goto('/navigation-lifecycle/before-navigate/prevent-navigation');
+		const pathname = '/navigation-lifecycle/before-navigate/prevent-navigation';
+		const current_url = baseURL + pathname;
 
-		await page.click('a[download]');
-		expect(await page.innerHTML('pre')).toBe('0 false undefined');
+		await page.goto(pathname);
 
-		await page.click('a[href="/navigation-lifecycle/before-navigate/a"]');
+		const download = page.waitForEvent('download');
+		await page.locator('a[download]').click();
+		await page.waitForURL(current_url);
+		await (await download).cancel();
 
-		expect(page.url()).toBe(baseURL + '/navigation-lifecycle/before-navigate/prevent-navigation');
-		expect(await page.innerHTML('pre')).toBe('1 false link');
+		expect(await page.locator('pre').innerText()).toBe('0 false undefined');
+
+		await page.locator('a[href="/navigation-lifecycle/before-navigate/a"]').click();
+		await page.waitForURL(current_url);
+
+		expect(await page.locator('pre').innerText()).toBe('1 false link');
```

Comment:
The `await` has to be outside the `expect` for this to work correctly, which I think is why this line is flaking. It should be something like:
```suggestion
		await expect(page.locator('pre')).toHaveText('1 false link');
```

Reply:
Thanks Ben! Seems like that did the trick